### PR TITLE
Hive obj output improvements

### PR DESCRIPTION
**Pull request by bimmlerd**
A few improvements to the hive obj graph. Mostly work by Jussi I'm shepherding through while we hack on other stuff.

### CODE REVIEW

File: pkg/hive/internal/reflect.go [hunk 1]
```diff
@@ -30,7 +31,25 @@ func FuncNameAndLocation(fn any) string {
...
+		return fmt.Sprintf("%s (%s:%d)", name, usefulPathSegment(file), line)
 	}
 	return name
 }
+
+// Purely a heuristic.
+var commonRoots = map[string]struct{}{
+	"pkg": {},
+	"cmd": {},
+}
```
Comment (giorio94) [inline pkg/hive/internal/reflect.go:43]: What about also adding `api`?

**Review by giorio94 — APPROVED** [review]
As far as I can tell, the PR looks good to me, just one nit inline. Unclear to me why I was pulled in as reviewer though, as I'm not part of sig-foundations :thinking:

**Review by dylandreimerink — APPROVED** [review]

File: pkg/hive/internal/reflect.go (hunk 1 above)
Comment (dylandreimerink) [reply pkg/hive/internal/reflect.go:43]: I don't think we have that many `api/*` objects in hive at the moment, do we? We can always add this later on.

**Pull request merged**